File: spring-graphql-webflux/src/test/java/com/example/demo/MutationTests.java
package com.example.demo;

import com.example.demo.gql.datafetchers.PostsDataFetcherController;
import com.example.demo.gql.types.CreatePostInput;
import com.example.demo.gql.types.Post;
import com.example.demo.service.AuthorService;
import com.example.demo.service.PostService;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.graphql.GraphQlTest;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.graphql.test.tester.GraphQlTester;
import reactor.core.publisher.Mono;

import java.util.Map;
import java.util.UUID;

import static org.mockito.Mockito.*;

@GraphQlTest(controllers = {PostsDataFetcherController.class})
@Slf4j
class MutationTests {

    @Autowired
    GraphQlTester graphQlTester;

    @MockBean
    PostService postService;

    @MockBean
    AuthorService authorService;

    @Test
    void createPosts() {
        when(postService.createPost(any(CreatePostInput.class))).thenReturn(Mono.just(
                Post.builder().id(UUID.randomUUID().toString())
                        .title("test title")
                        .content("test content")
                        .build()
        ));

        var query = "mutation createPost($input: CreatePostInput!){createPost(createPostInput:$input){id, title, content}}";
        graphQlTester.document(query)
                .variable("input", Map.of(
                        "title", "test title",
                        "content", "test content"
                ))
                .execute()
                .path("data.createPost.title")
                .entity(String.class)
                .isEqualTo("test title");

        verify(postService, times(1)).createPost(any(CreatePostInput.class));
        verifyNoMoreInteractions(postService);
    }

}

File: spring-graphql-rsocket-kotlin-co/src/test/kotlin/com/example/demo/MutationTests.kt
package com.example.demo

import com.example.demo.gql.datafetchers.PostController
import com.example.demo.gql.types.Post
import com.ninjasquad.springmockk.MockkBean
import io.mockk.coEvery
import io.mockk.coVerify
import kotlinx.coroutines.test.runTest
import org.junit.jupiter.api.Test
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.graphql.GraphQlTest
import org.springframework.graphql.test.tester.GraphQlTester
import java.time.LocalDateTime
import java.util.*

@GraphQlTest(controllers = [PostController::class])
internal class MutationTests {
    companion object {
        private val log = LoggerFactory.getLogger(MutationTests::class.java)
    }

    @Autowired
    lateinit var graphQlTester: GraphQlTester

    @MockkBean
    lateinit var postService: PostService

    @MockkBean
    lateinit var authorService: AuthorService

    @Test
    fun createPosts() = runTest {
        coEvery { postService.createPost(any()) } returns
                Post(
                    id = UUID.randomUUID(),
                    title = "test title",
                    content = "test content",
                    status = PostStatus.DRAFT,
                    createdAt = LocalDateTime.now()
                )

        val inputHolder = "\$input"
        val query = """
                mutation createPost($inputHolder: CreatePostInput!){
                    createPost(createPostInput:$inputHolder){
                        id, 
                        title, 
                        content
                    }
                }
            """.trimIndent()
        graphQlTester.document(query)
            .variable(
                "input",
                mapOf(
                    "title" to "test title",
                    "content" to "test content"
                )
            )
            .execute()
            .path("data.createPost.title")
            .entity(String::class.java)
            .isEqualTo("TEST TITLE")

        coVerify(exactly = 1) { postService.createPost(any()) }
    }
}
